src/cookbooks/client_guidelines.md

```diff
@@ -0,0 +1,137 @@
+
+## Extensions
+
+Determining whether an extension is appropriate for an experiment will be decided on a case-by-case basis. As in the “Feature Gates” section above, landing features directly in-tree is preferred to using an extension for experiments.
+
+__Each extension can represent a different experimental treatment (preferred), or the extension can choose the branch internally__.
+
+Firefox currently supports the [Web Extensions API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions).
+
+__If new WebExtension APIs are needed, they should land in-tree__. Extensions which are signed by Mozilla can load privileged code using the [WebExtension Experiments](https://firefox-source-docs.mozilla.org/toolkit/components/extensions/webextensions/index.html), but this is not preferred.
+
+WebExtensions go through the same correctness and performance tests as other features. This is possible using the Mozilla tryserver by dropping your XPI into `testing/profiles/common/extensions` in `mozilla-central` and pushing to Tryserver - see the [Testing Extensions](#testing-extensions) section below.
+
+### SHIELD studies
+
+The previous version of the experiments program, SHIELD, always bundled privileged code with extensions and would do things such as mock UI features in Firefox.
+
+This sort of approach is discouraged for new features - land these (or the necessary WebExtension APIs) in-tree instead.
+
+For the moment, the [SHIELD Study Add-on Utilities](https://github.com/mozilla/shield-studies-addon-utils/) may be used if the extension needs to control the lifecycle of the study, but using one extension per experimental treatment makes this unnecessary and is preferred. The APIs provided by the SHIELD Study Add-on Utilities will be available as privileged APIs shipped with Firefox soon.
+
+
+# Development and Testing
+## Testing Built-in Features
+
+Firefox features go through standard development and testing processes. See the [Firefox developer guide][firefox-dev-docs] for more information.
+
+
+## Testing Extensions
+
+Extensions do not need to go through the same process, but should take advantage of Mozilla CI and bug tracking systems:
+
+1. File a Bugzilla bug.
+2. Use the Mozilla CI to test changes (tryserver).
+3. Performance tests (__this step is required__) - extension XPI files should be placed in `testing/profiles/common/extensions/`, which will cause test harnesses to load the XPI.
+4. Custom unit/functional tests (AKA `xpcshell`/`mochitest`) may be placed in `testing/extensions`, although running these tests outside Mozilla CI is acceptable so these are __optional__.
+5. Receive reviewer approval. A Firefox peer __must sign off__ if this extension contains privileged code, aka WebExtension Experiments.
+6. Extension is signed.
```

Comment:
The process around this isn't wonderful. There's been a lot of talk about automating signing but it's just not there yet. I'd rather leave it vague for now, since i think Experimenter and PMs will keep things moving for now, but that's not a great solution.